learn/cookbooks/multitenancy_nodejs.mdx

```diff
@@ -0,0 +1,214 @@
+- A search API key — available in your Meilisearch dashboard
+- A search API key UID — retrieve it using the [keys endpoints](/reference/api/keys)
+
+<Capsule intent="tip">
+Prefer self-hosting? Read our [installation guide](/learn/getting_started/installation).
+</Capsule>
+
+## Data models
+
+In this guide, we'll use a simple data model to represent medical appointments. The documents in the Meilisearch index will look like this:
+
+```json
+[
+  {
+    "id": 1,
+    "patient": "John",
+    "details": "I think I caught a cold. Can you help me?",
+    "status": "pending"
+  },
+  {
+    "id": 2,
+    "patient": "Zia",
+    "details": "I'm suffering from fever. I need an appointment ASAP.",
+    "status": "pending"
+  },
+  {
+    "id": 3,
+    "patient": "Kevin",
+    "details": "Some confidential information Kevin has shared.",
+    "status": "confirmed"
+  }
+]
+```
+
+For the purpose of this guide, we assume documents are stored in an `appointments` index.
+
+## Creating a tenant token
+
+You will generate a tenant token that will allow a given patient to search only for their appointments. To achieve this, you will create a tenant token that filters the results based on the patient's ID.
+
+In a `search.js` file, you can create the tenant token like this:
+
+```js
+// search.js
+
+import { Meilisearch } from 'meilisearch'
+
+const apiKey = '<YOUR_SEARCH_API_KEY>'
+const apiKeyUid = '<YOUR_SEARCH_API_KEY_UID>'
+const indexName = 'appointments'
+
+const client = new Meilisearch({
+	host: 'https://edge.meilisearch.com', // Your Meilisearch host
+	apiKey: apiKey
+})
+
+export function createTenantToken(patientName) {
+  const searchRules = {
+    [indexName]: {
+      'filter': `patient = ${patientName}`
+    }
+  }
+
+  const tenantToken = client.generateTenantToken(
+    apiKeyUid,
+    searchRules,
+    {
+      expiresAt: new Date('2030-01-01'), // Choose an expiration date
+      apiKey: apiKey,
+    }
+  )
+  return tenantToken
+}
+```
+
+When Meilisearch gets a search query with a tenant token, it decodes it and applies the search rules to the search request. In this example, the results are filtered by the `patient` field. This means that a patient can only search for their own appointments.
+
+Next, you will use this tenant token to perform searches against your Meilisearch instance.
+
+## Using the tenant token
+
+Now that you have a tenant token, you will use it to perform searches. Like all search API keys, you can safely use it in client-side code.
+
+To achieve this, you will need to:
+- On the server—Create an endpoint to send the token to your front-end
+- On the client—Retrieve the token and use it to perform searches
+
+### Serving the tenant token
+
+This guide uses [Express.js](https://expressjs.com/en/starter/installing.html) to create a server. You can install `express` by running:
+
+```sh
+# with NPM
+npm i express
+# with Yarn
+yarn add express
+# with pnpm
+pnpm add express
```

Comment:
I think it would be ok if you picked one package manager and removed the others. This way users don't need to make a choice.

Reply:
I feel like this is less and less common in the JS world, and the standard is to remind people that they can use different package managers. I will leave it like this for now, but we can definitely revisit later.